adminapi/request.py

```diff
@@ -17,30 +17,40 @@
 except ImportError:
     import json
 
-BASE_URL = os.environ.get(
-    'SERVERADMIN_BASE_URL',
-    'https://serveradmin.innogames.de/api'
-)
+
+class Settings:
+    base_url = os.environ.get(
+        'SERVERADMIN_BASE_URL',
+        'https://serveradmin.innogames.de/api'
+    )
+    timeout = 60
+    tries = 3
+    auth_token = None
 
 
 def calc_security_token(auth_token, timestamp, content):
     message = str(timestamp) + ':' + str(content)
     return hmac.new(
-        auth_token.encode('utf8'), message.encode('utf8'), hashlib.sha1
+        auth_token.encode('utf8'), message.encode('utf8'), sha1
     ).hexdigest()
 
 
-def send_request(endpoint, data, auth_token, timeout=None):
-    if not auth_token:
-        auth_token = get_auth_token()
+def calc_app_id(auth_token):
+    return sha1(auth_token.encode('utf8')).hexdigest()
+
+
+def send_request(endpoint, data):
+    if not Settings.auth_token:
+        Settings.auth_token = get_auth_token()
 
     data_json = json.dumps(data, default=json_encode_extra)
 
-    for retry in reversed(range(3)):
+    for retry in reversed(range(Settings.tries)):
         try:
-            req = _build_request(endpoint, auth_token, data_json)
+            req = _build_request(endpoint, Settings.auth_token, data_json)
             return json.loads(
-                urlopen(req, timeout=timeout).read().decode('utf8'))
+                urlopen(req, timeout=Settings.timeout).read().decode('utf8')
+            )
         except HTTPError as error:
             if error.code < 500:
                 raise
```

Comment:
Here too, defensive programming would go for `retry <= 0` below.

Reply:
Wonder if the `time.sleep` should also be configurable via `Settings`.

Reply:
Done both